src/Controls/src/Core/Layout/Layout.cs

```diff
@@ -9,6 +9,144 @@
 
 namespace Microsoft.Maui.Controls
 {
+
+#pragma warning disable RS0016 // Add public types and members to the declared API
+	public abstract class Layout<T> : Layout where T : View, ICrossPlatformLayout
+	{
+		public new IList<T> Children { get; }
+		public Layout()
+		{
+			Children = new CastingList<T, IView>(this);
+		}
+
+		// If we want to match the APIS
+
+		public void ForceLayout() => SizeAllocated(Width, Height);
+
+		protected override void OnChildAdded(Element child)
+		{
+			base.OnChildAdded(child);
+
+			if (child is T typedChild)
+			{
+				OnAdded(typedChild);
+#pragma warning disable CS0618 // Type or member is obsolete
+				typedChild.MeasureInvalidated += OnChildMeasureInvalidated;
+#pragma warning restore CS0618 // Type or member is obsolete
+			}
+		}
+
+		protected override void OnChildRemoved(Element child, int oldLogicalIndex)
+		{
+			base.OnChildRemoved(child, oldLogicalIndex);
+
+			if (child is T typedChild)
+			{
+				OnRemoved(typedChild);
+#pragma warning disable CS0618 // Type or member is obsolete
+				typedChild.MeasureInvalidated -= OnChildMeasureInvalidated;
+#pragma warning restore CS0618 // Type or member is obsolete
+			}
+		}
+
+		/// <summary>
+		/// Invoked when a child is added to the layout. Implement this method to add class handling for this event.
+		/// </summary>
+		/// <param name="view">The view which was added.</param>
+		protected virtual void OnAdded(T view)
+		{
+		}
+
+		/// <summary>
+		/// Invoked when a child is removed the layout. Implement this method to add class handling for this event.
+		/// </summary>
+		/// <param name="view">The view which was removed.</param>
+		protected virtual void OnRemoved(T view)
+		{
+		}
+
+		[Obsolete("Use ZIndex")]
+		public void LowerChild(View view)
+		{
+			if (!this.Contains(view) || this.First() == view)
+			{
+				return;
+			}
+
+			this.Remove(view);
+			this.Insert(view, 0);
+			OnChildrenReordered();
+		}
+
+		[Obsolete("Use ZIndex")]
+		public void RaiseChild(View view)
+		{
+			if (!this.Contains(view) || this.Last() == view)
+			{
+				return;
+			}
+
+			this.Remove(view);
+			this.Insert(view,this.Count - 1);
+			OnChildrenReordered();
+		}
+
+		/// <summary>
+		/// Instructs the layout to relayout all of its children.
+		/// </summary>
+		/// <remarks>This method starts a new layout cycle for the layout. Invoking this method frequently can negatively impact performance.</remarks>
+		protected virtual void InvalidateLayout()
+		{
+			// Todo call this when InvalidateMeasure is called?
+			(this as IView).InvalidateMeasure();
+		}
+
+		Graphics.Size _layoutChildren = Graphics.Size.Zero;
```

Comment:
This field name is a bit confusing as it sounds like a command. Maybe this is better as `_lastArrangeBounds`?